### ISSUE TITLE
Crash in -getProperty when there's a missing import

### ISSUE THREAD

**Issue by KirillOsenkov** [dotnet/msbuild#9691]
`msbuild -getProperty:Username` on any project that has a missing import:

```
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
Microsoft.Build.Exceptions.InvalidProjectFileException: The imported project "C:\temp\NonExisting.proj" was not found. Confirm that the expression in the Import declaration "NonExisting.proj" is correct, and that the file exists on disk.  C:\temp\1.proj
   at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject(String errorSubCategoryResourceName, IElementLocation elementLocation, String resourceName, Object[] args)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImportsFromUnescapedImportExpression(String directoryOfImportingFile, ProjectImportElement importElement, String unescapedExpression, Boolean throwOnFileNotExistsError, List`1& imports)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImportsFromUnescapedImportExpressionConditioned(String directoryOfImportingFile, ProjectImportElement importElement, List`1& projects, SdkResult& sdkResult)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImports(String directoryOfImportingFile, ProjectImportElement importElement, SdkResult& sdkResult)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate()
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate(IEvaluatorData`4 data, Project project, ProjectRootElement root, ProjectLoadSettings loadSettings, Int32 maxNodeCount, PropertyDictionary`1 environmentProperties, ILoggingService loggingService, IItemFactory`2 itemFactory, IToolsetProvider toolsetProvider, IDirectoryCacheFactory directoryCacheFactory, ProjectRootElementCacheBase projectRootElementCache, BuildEventContext buildEventContext, ISdkResolverService sdkResolverService, Int32 submissionId, EvaluationContext evaluationContext, Boolean interactive)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext, Boolean interactive)
   at Microsoft.Build.Evaluation.Project..ctor(String projectFile, IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext, IDirectoryCacheFactory directoryCacheFactory, Boolean interactive)
   at Microsoft.Build.Evaluation.ProjectCollection.LoadProject(String fileName, IDictionary`2 globalProperties, String toolsVersion)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String commandLine)
```

**Comment by KirillOsenkov** [comment]
@ladipro

**Comment by KirillOsenkov** [comment]
Probably need to set ignore missing imports and friends??

```
                var projectLoadSettings =
                    ProjectLoadSettings.IgnoreMissingImports |
                    ProjectLoadSettings.IgnoreEmptyImports |
                    ProjectLoadSettings.IgnoreInvalidImports |
                    ProjectLoadSettings.DoNotEvaluateElementsWithFalseCondition |
                    ProjectLoadSettings.RecordEvaluatedItemElements;
```

**Comment by AR-May** [comment]
@Forgind fyi

**Comment by rainersigwald** [comment]
> Probably need to set ignore missing imports and friends??

I don't think we should take this approach, but make the error cleaner (like what happens when you run a build). We wouldn't want to return a _wrong_ result (undefined properties when an import can't be found).

**Comment by Forgind** [comment]
What version are you using? I'm like 95% sure this was fixed by:
https://github.com/dotnet/msbuild/pull/9400

I tried building current main just now and failed to reproduce this using a simple console app with a missing import.

**Comment by Forgind** [comment]
These were my results:
```
>C:\GitHub\msbuild\artifacts\bin\bootstrap\net472\MSBuild\Current\Bin\MSBuild.exe temp.csproj -getProperty:Username <pathTo>\temp.csproj(11,3): error MSB4019: The imported project "<pathTo>\notReal.csproj" was not found. Confirm that the expression in the Import declaration "notReal.csproj" is correct, and that the file exists on disk.
```

and

```
>C:\GitHub\msbuild\artifacts\bin\bootstrap\net472\MSBuild\Current\Bin\MSBuild.exe temp.csproj -getProperty:Username -t:Foo
<pathTo>\temp.csproj(11,3): error MSB4019: The imported project "<pathTo>\notReal.csproj" was not found. Confirm that the expression in the Import declaration "notReal.csproj" is correct, and that the file exists on disk.
Build failed. Properties, Items, and Target results cannot be obtained. See details in stderr above.
```

That all looks right to me.

**Comment by Forgind** [comment]
Closing as fixed via #9400
